tutorials/tts/FastPitch_Speaker_Interpolation.ipynb

```diff
@@ -0,0 +1,436 @@
+    "    speaker_emb_2 = model.fastpitch.speaker_emb(torch.tensor(speaker_2, dtype=torch.int32).cuda()).clone().detach()\n",
+    "    return weight_speaker_1 * speaker_emb_1 + weight_speaker_2 * speaker_emb_2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e542bd19",
+   "metadata": {},
+   "source": [
+    "**Generate Speech using speaker interpolation**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae9d5f65",
+   "metadata": {},
+   "source": [
+    "Now we will use this process of speaker interpolation to generate speech using original speaker voices and interpolated speaker voice. For this we will need a vocoder to convert generated spectrogram to waveform speech. So let's load HiFiGAN vocoder."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "88f7544e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vocoder = HifiGanModel.from_pretrained(\"tts_en_hifitts_hifigan_ft_fastpitch\")\n",
+    "vocoder.cuda().eval()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2dcf5ea8",
+   "metadata": {},
+   "source": [
+    "Now we will define a function which will take as input the spectrogram generator model, vocoder model, text to be converted to speech and speaker id. This function will return spectrogram and audio of the generated speech."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9b6e10fd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(spec_gen_model, vocoder_model, str_input, speaker=None):\n",
+    "    \"\"\"\n",
+    "    Synthesizes spectrogram and audio from a text string given a spectrogram synthesis and vocoder model.\n",
+    "    \n",
+    "    Args:\n",
+    "        spec_gen_model: Spectrogram generator model (FastPitch in our case)\n",
+    "        vocoder_model: Vocoder model (HiFiGAN in our case)\n",
+    "        str_input: Text input for the synthesis\n",
+    "        speaker: Speaker ID\n",
+    "    \n",
+    "    Returns:\n",
+    "        spectrogram and waveform of the synthesized audio.\n",
+    "    \"\"\"\n",
+    "    spec_gen_model.eval()\n",
+    "    with torch.no_grad():\n",
+    "        parsed = spec_gen_model.parse(str_input)\n",
+    "        if speaker is not None:\n",
+    "            speaker = torch.tensor([speaker]).long().to(device=spec_gen_model.device)\n",
+    "        spectrogram = spec_gen_model.generate_spectrogram(tokens=parsed, speaker=speaker)\n",
+    "        audio = vocoder_model.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "        \n",
+    "    if spectrogram is not None:\n",
+    "        if isinstance(spectrogram, torch.Tensor):\n",
+    "            spectrogram = spectrogram.to('cpu').numpy()\n",
+    "        if len(spectrogram.shape) == 3:\n",
+    "            spectrogram = spectrogram[0]\n",
+    "    if isinstance(audio, torch.Tensor):\n",
+    "        audio = audio.to('cpu').numpy()\n",
```

Comment:
How does this workflow yield a spectrogram which is None or a non-tensor?

Similar question for the other "if" conditions.